[ Site Isolation ] Add iOS site-isolation post-commit bots

robert-jenner opened the pull request:
#### 8d32d0fb851dd306c4c63e9fd3d7fe99246be680
<pre>
[ Site Isolation ] Add iOS site-isolation post-commit bots
<a href="https://rdar.apple.com/164646440">rdar://164646440</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=302481">https://bugs.webkit.org/show_bug.cgi?id=302481</a>

Reviewed by Sihui Liu.

This changes brings up post-commit testing for site-isolation on iOS. It will run layout-tests, and API-Tests, and report as its own site-isolation flavor.

* LayoutTests/platform/ios-site-isolation/TestExpectations: Added.
* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(_set_up_derived_options):

Canonical link: <a href="https://commits.webkit.org/303017@main">https://commits.webkit.org/303017@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a2be328b6244bc36617b1ad848cede6809fc175b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/130902 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3226 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/41861 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138330 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/82563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/e78ea530-93ff-47c4-bb75-8c5c372ff20c) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/132773 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3208 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3068 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/99736 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/82563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/16cd2fa2-ec27-475f-8594-03ab2db16175) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/133848 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/2313 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117207 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/80438 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/abb71188-4fb1-4df6-b1f9-8b391c230536) 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/130251 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/2230 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/35339 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/81589 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/110820 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/35841 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/140812 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/2969 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/2671 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/108251 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/130331 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3015 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113538 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/108176 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27534 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2266 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/31964 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56006 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3037 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/2858 "Built successfully") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/3058 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/2967 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

bls1999 commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -501,6 +501,17 @@ def _set_up_derived_options(port, options):
             raise RuntimeError('--site-isolation implicitly sets the result flavor, this should not be overridden')
         options.result_report_flavor = 'site-isolation'
 
+    if port.port_name == "ios" and options.site_isolation:
```
Will `port.port_name` always be `ios`? I'm thinking in the case of iPad, iOS simulator, etc.

Maybe something like:
```
if (port.port_name.startswith('ios') or port.port_name.startswith('ipad')) and ...
```

robert-jenner replied:
That's valid. Especially since they likely won't be testing on device. I'll make this change.

bls1999 replied:
Thanks for resolving!

robert-jenner replied:
We have discussed that we are leaving iPad out of this at this time. I have made a change that should satisfy this comment. So I'm marking as resolved.

bls1999 reviewed (commented):
1 comment, if that's fine then [unofficial] r=me.

JonWBedard commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -501,6 +501,17 @@ def _set_up_derived_options(port, options):
         options.result_report_flavor = 'site-isolation'
 
+    if port.port_name == "ios" and options.site_isolation:
+        host = Host()
```
`port` object has a host, should be `port.host.scm()` instead of constructing a new `host` object.

robert-jenner replied:
I believe I have addressed this in my most recent revision.

bls1999 reviewed (approved):
[unofficial] r=me after feedback addressed, thanks @robert-jenner!

JonWBedard commented on Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
```diff
@@ -501,6 +501,16 @@ def _set_up_derived_options(port, options):
         options.result_report_flavor = 'site-isolation'
 
+    if (port.port_name.startswith('ios')) and options.site_isolation:
+        port.host.scm().checkout_root
```
This line doesn't do anything.

robert-jenner replied:
Removed line 505. This is good to land.

robert-jenner committed:
[ Site Isolation ] Add iOS site-isolation post-commit bots
rdar://164646440
https://bugs.webkit.org/show_bug.cgi?id=302481

Reviewed by Sihui Liu.

This changes brings up post-commit testing for site-isolation on iOS. It will run layout-tests, and API-Tests, and report as its own site-isolation flavor.

* LayoutTests/platform/ios-site-isolation/TestExpectations: Added.
* Tools/CISupport/build-webkit-org/config.json:
* Tools/CISupport/build-webkit-org/factories_unittest.py:
(TestExpectedBuildSteps):
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests.py:
(_set_up_derived_options):

Canonical link: https://commits.webkit.org/303017@main
